### PR TITLE
Fixed several permission problems for recent android versions.

### PR DESCRIPTION
**Pull request by dov**
* Added necessary new permissions to AndroidManifest.xml file.
* Start the DeviceScanActivity by querying and requesting these run
  time permissions.

Signed-off-by: Dov Grobgeld <dov.grobgeld@gmail.com>

### CODE REVIEW

File: Application/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -32,6 +32,8 @@
 
     <uses-permission android:name="android.permission.BLUETOOTH"/>
     <uses-permission android:name="android.permission.BLUETOOTH_ADMIN"/>
+    <uses-permission android:name="android.permission.ACCESS_COARSE_LOCATION"/>
+    <uses-permission android:name="android.permission.ACCESS_FINE_LOCATION"/>
```
Comment (tonnenpinguin) [inline Application/src/main/AndroidManifest.xml:36]: What do you need the fine location for?

File: Application/src/main/java/com/example/android/bluetoothlegatt/DeviceScanActivity.java [hunk 2]
```diff
@@ -35,7 +37,7 @@
 import android.widget.ListView;
 import android.widget.TextView;
 import android.widget.Toast;
-
+import android.util.Log;
```
Comment (tonnenpinguin) [inline Application/src/main/java/com/example/android/bluetoothlegatt/DeviceScanActivity.java:40]: Unused import?

File: Application/src/main/AndroidManifest.xml (hunk 1 above)
Comment (sebastienrouif) [reply Application/src/main/AndroidManifest.xml:36]: check the note at the end or this section : https://developer.android.com/guide/topics/connectivity/bluetooth-le.html#permissions

Comment (neuhaus) [reply Application/src/main/AndroidManifest.xml:36]: Isn't the ACCESS_COARSE_LOCATION sufficient then?

**Comment by codingjeremy** [comment]
I am closing this issue/PR, as it has been migrated to the new repo linked above in the comments. Thank you!

**Pull request closed without merging**